src/Engine/Graphics/Renderer/OpenGLRenderer.cpp

```diff
@@ -3847,732 +3853,732 @@ void OpenGLRenderer::DrawIndoorFaces() {
-                if (pface->uSectorID == listsector) {
-                    onlist = true;
-                    break;
-                }
-            }
-            if (!onlist) continue;
+    if (!decal_builder->bloodsplat_container->uNumBloodsplats) return;
+    static RenderVertexSoft static_vertices_buff_in[64];  // buff in
 
+    // loop over faces
+    for (int test = 0; test < pIndoor->pFaces.size(); test++) {
+        BLVFace *pface = &pIndoor->pFaces[test];
 
-            decal_builder->ApplyBloodsplatDecals_IndoorFace(test);
-            if (!decal_builder->uNumSplatsThisFace) continue;
+        if (pface->isPortal()) continue;
+        if (!pface->GetTexture()) continue;
 
-            // copy to buff in
-            for (unsigned i = 0; i < pface->uNumVertices; ++i) {
-                static_vertices_buff_in[i].vWorldPosition.x =
-                    pIndoor->pVertices[pface->pVertexIDs[i]].x;
-                static_vertices_buff_in[i].vWorldPosition.y =
-                    pIndoor->pVertices[pface->pVertexIDs[i]].y;
-                static_vertices_buff_in[i].vWorldPosition.z =
-                    pIndoor->pVertices[pface->pVertexIDs[i]].z;
-                static_vertices_buff_in[i].u = (signed short)pface->pVertexUIDs[i];
-                static_vertices_buff_in[i].v = (signed short)pface->pVertexVIDs[i];
+        // check if faces is visible
+        bool onlist = false;
+        for (unsigned i = 0; i < pBspRenderer->uNumVisibleNotEmptySectors; ++i) {
+            int listsector = pBspRenderer->pVisibleSectorIDs_toDrawDecorsActorsEtcFrom[i];
+            if (pface->uSectorID == listsector) {
+                onlist = true;
+                break;
             }
-
-            // blood draw
-            decal_builder->BuildAndApplyDecals(uCurrentAmbientLightLevel, LocationIndoors, pface->facePlane,
-                pface->uNumVertices, static_vertices_buff_in,
-                0, pface->uSectorID);
+        }
+        if (!onlist) continue;
+
+
+        decal_builder->ApplyBloodsplatDecals_IndoorFace(test);
+        if (!decal_builder->uNumSplatsThisFace) continue;
+
+        // copy to buff in
+        for (unsigned i = 0; i < pface->uNumVertices; ++i) {
+            static_vertices_buff_in[i].vWorldPosition.x =
+                pIndoor->pVertices[pface->pVertexIDs[i]].x;
+            static_vertices_buff_in[i].vWorldPosition.y =
+                pIndoor->pVertices[pface->pVertexIDs[i]].y;
+            static_vertices_buff_in[i].vWorldPosition.z =
+                pIndoor->pVertices[pface->pVertexIDs[i]].z;
+            static_vertices_buff_in[i].u = (signed short)pface->pVertexUIDs[i];
+            static_vertices_buff_in[i].v = (signed short)pface->pVertexVIDs[i];
         }
 
+        // blood draw
+        decal_builder->BuildAndApplyDecals(uCurrentAmbientLightLevel, LocationIndoors, pface->facePlane,
+            pface->uNumVertices, static_vertices_buff_in,
+            0, pface->uSectorID);
+    }
```

Comment:
What's up with all these changes? X)

Are those just code formatting?

If so, let's not mix code format changes with real changes. It's hard to review when they are mixed up.